File: app/src/main/java/com/battlelancer/seriesguide/model/SgShow.java
package com.battlelancer.seriesguide.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.battlelancer.seriesguide.provider.SeriesGuideContract.Shows;
import com.battlelancer.seriesguide.provider.SeriesGuideDatabase.Tables;

@Entity(tableName = Tables.SHOWS)
public class SgShow {

    @PrimaryKey
    @ColumnInfo(name = Shows._ID)
    public int tvdbId;

    @ColumnInfo(name = Shows.SLUG)
    public String slug = "";

    /**
     * Ensure this is NOT null (enforced through database constraint).
     */
    @ColumnInfo(name = Shows.TITLE)
    @NonNull
    public String title = "";

    /**
     * The title without any articles (e.g. 'the' or 'an'). Added with db version 33.
     */
    @ColumnInfo(name = Shows.TITLE_NOARTICLE)
    public String titleNoArticle;

    @ColumnInfo(name = Shows.OVERVIEW)
    public String overview = "";

    /**
     * Local release time. Encoded as integer (hhmm).
     *
     * <pre>
     * Example: 2035
     * Default: -1
     * </pre>
     */
    @ColumnInfo(name = Shows.RELEASE_TIME)
    public Integer releaseTime;
    /**
     * Local release week day. Encoded as integer.
     * <pre>
     * Range:   1-7
     * Daily:   0
     * Default: -1
     * </pre>
     */
    @ColumnInfo(name = Shows.RELEASE_WEEKDAY)
    public Integer releaseWeekDay;
    @ColumnInfo(name = Shows.RELEASE_COUNTRY)
    public String releaseCountry;
    @ColumnInfo(name = Shows.RELEASE_TIMEZONE)
    public String releaseTimeZone;

    @ColumnInfo(name = Shows.FIRST_RELEASE)
    public String firstRelease;

    @ColumnInfo(name = Shows.GENRES)
    public String genres = "";
    @ColumnInfo(name = Shows.NETWORK)
    public String network = "";

    @ColumnInfo(name = Shows.RATING_GLOBAL)
    public Double ratingGlobal;
    @ColumnInfo(name = Shows.RATING_VOTES)
    public Integer ratingVotes;
    @ColumnInfo(name = Shows.RATING_USER)
    public Integer ratingUser;

    @ColumnInfo(name = Shows.RUNTIME)
    public String runtime = "";
    @ColumnInfo(name = Shows.STATUS)
    public String status = "";
    @ColumnInfo(name = Shows.CONTENTRATING)
    public String contentRating = "";

    @ColumnInfo(name = Shows.NEXTEPISODE)
    public String nextEpisode = "";

    @ColumnInfo(name = Shows.POSTER)
    public String poster = "";

    @ColumnInfo(name = Shows.POSTER_SMALL)
    public String posterSmall = "";

    @ColumnInfo(name = Shows.NEXTAIRDATEMS)
    public Long nextAirdateMs;
    @ColumnInfo(name = Shows.NEXTTEXT)
    public String nextText = "";

    @ColumnInfo(name = Shows.IMDBID)
    public String imdbId = "";
    @ColumnInfo(name = Shows.TRAKT_ID)
    public Integer traktId = 0;

    @ColumnInfo(name = Shows.FAVORITE)
    public boolean favorite = false;

    @ColumnInfo(name = Shows.HEXAGON_MERGE_COMPLETE)
    public boolean hexagonMergeComplete = true;

    @ColumnInfo(name = Shows.HIDDEN)
    public boolean hidden = false;

    @ColumnInfo(name = Shows.LASTUPDATED)
    public long lastUpdatedMs = 0L;
    @ColumnInfo(name = Shows.LASTEDIT)
    public long lastEditedSec = 0L;

    @ColumnInfo(name = Shows.LASTWATCHEDID)
    public int lastWatchedEpisodeId = 0;
    @ColumnInfo(name = Shows.LASTWATCHED_MS)
    public long lastWatchedMs = 0L;

    @ColumnInfo(name = Shows.LANGUAGE)
    public String language = "";

    @ColumnInfo(name = Shows.UNWATCHED_COUNT)
    public int unwatchedCount = SgShow2.UNKNOWN_UNWATCHED_COUNT;

    @ColumnInfo(name = Shows.NOTIFY)
    public boolean notify = true;
}

File: app/src/main/java/com/battlelancer/seriesguide/shows/database/SgShow.kt
// SPDX-License-Identifier: Apache-2.0
// Copyright 2018-2025 Uwe Trottmann

package com.battlelancer.seriesguide.shows.database

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.battlelancer.seriesguide.provider.SeriesGuideContract.Shows
import com.battlelancer.seriesguide.provider.SeriesGuideDatabase.Tables

/**
 * Legacy show entity kept for migration of legacy data. See [SgShow2].
 */
@Entity(tableName = Tables.SHOWS)
data class SgShow(
    @PrimaryKey
    @ColumnInfo(name = Shows._ID)
    var tvdbId: Int,

    @ColumnInfo(name = Shows.SLUG)
    var slug: String? = "",

    /**
     * Ensure this is NOT null (enforced through database constraint).
     */
    @ColumnInfo(name = Shows.TITLE)
    var title: String = "",

    /**
     * The title without any articles (e.g. 'the' or 'an'). Added with db version 33.
     */
    @ColumnInfo(name = Shows.TITLE_NOARTICLE)
    var titleNoArticle: String? = null,

    @ColumnInfo(name = Shows.OVERVIEW)
    var overview: String? = "",

    /**
     * Local release time. Encoded as integer (hhmm).
     *
     * ```
     * Example: 2035
     * Default: -1
     * ```
     */
    @ColumnInfo(name = Shows.RELEASE_TIME)
    var releaseTime: Int? = null,
    /**
     * Local release week day. Encoded as integer.
     *
     * ```
     * Range:   1-7
     * Daily:   0
     * Default: -1
     * ```
     */
    @ColumnInfo(name = Shows.RELEASE_WEEKDAY)
    var releaseWeekDay: Int? = null,
    @ColumnInfo(name = Shows.RELEASE_COUNTRY)
    var releaseCountry: String? = null,
    @ColumnInfo(name = Shows.RELEASE_TIMEZONE)
    var releaseTimeZone: String? = null,

    @ColumnInfo(name = Shows.FIRST_RELEASE)
    var firstRelease: String? = null,

    @ColumnInfo(name = Shows.GENRES)
    var genres: String? = "",
    @ColumnInfo(name = Shows.NETWORK)
    var network: String? = "",

    @ColumnInfo(name = Shows.RATING_GLOBAL)
    var ratingGlobal: Double? = null,
    @ColumnInfo(name = Shows.RATING_VOTES)
    var ratingVotes: Int? = null,
    @ColumnInfo(name = Shows.RATING_USER)
    var ratingUser: Int? = null,

    @ColumnInfo(name = Shows.RUNTIME)
    var runtime: String? = "",
    @ColumnInfo(name = Shows.STATUS)
    var status: String? = "",
    @ColumnInfo(name = Shows.CONTENTRATING)
    var contentRating: String? = "",

    @ColumnInfo(name = Shows.NEXTEPISODE)
    var nextEpisode: String? = "",

    @ColumnInfo(name = Shows.POSTER)
    var poster: String? = "",

    @ColumnInfo(name = Shows.POSTER_SMALL)
    var posterSmall: String? = "",

    @ColumnInfo(name = Shows.NEXTAIRDATEMS)
    var nextAirdateMs: Long? = null,
    @ColumnInfo(name = Shows.NEXTTEXT)
    var nextText: String? = "",

    @ColumnInfo(name = Shows.IMDBID)
    var imdbId: String? = "",
    @ColumnInfo(name = Shows.TRAKT_ID)
    var traktId: Int? = 0,

    @ColumnInfo(name = Shows.FAVORITE)
    var favorite: Boolean = false,

    @ColumnInfo(name = Shows.HEXAGON_MERGE_COMPLETE)
    var hexagonMergeComplete: Boolean = true,

    @ColumnInfo(name = Shows.HIDDEN)
    var hidden: Boolean = false,

    @ColumnInfo(name = Shows.LASTUPDATED)
    var lastUpdatedMs: Long = 0L,
    @ColumnInfo(name = Shows.LASTEDIT)
    var lastEditedSec: Long = 0L,

    @ColumnInfo(name = Shows.LASTWATCHEDID)
    var lastWatchedEpisodeId: Int = 0,
    @ColumnInfo(name = Shows.LASTWATCHED_MS)
    var lastWatchedMs: Long = 0L,

    @ColumnInfo(name = Shows.LANGUAGE)
    var language: String? = "",

    @ColumnInfo(name = Shows.UNWATCHED_COUNT)
    var unwatchedCount: Int = SgShow2.UNKNOWN_UNWATCHED_COUNT,

    @ColumnInfo(name = Shows.NOTIFY)
    var notify: Boolean = true
)
